config/openiotsdk/CMakeLists.txt

```diff
@@ -216,17 +217,28 @@ ExternalProject_Add(
     PREFIX                  ${CMAKE_CURRENT_BINARY_DIR}
     SOURCE_DIR              ${CHIP_ROOT}
     BINARY_DIR              ${CMAKE_CURRENT_BINARY_DIR}
-    CONFIGURE_COMMAND       ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/make_gn_args.py
-                                @args.tmp > args.gn &&
-                            ${GN_EXECUTABLE}
+    CONFIGURE_COMMAND       ""
+    CONFIGURE_HANDLED_BY_BUILD TRUE
+    BUILD_COMMAND           ${CMAKE_COMMAND} -E echo "Starting Matter library build in ${CMAKE_CURRENT_BINARY_DIR}"
+    COMMAND                 ${Python3_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/make_gn_args.py @args.tmp > args.gn.tmp
+    # Replace the config if it has changed
+    COMMAND                 bash -c "(! diff -q args.gn.tmp args.gn && mv args.gn.tmp args.gn) || true" 
```

Comment:
Why do we need to check this?

Reply:
Replacing the config with a new one; even if unchanged seems to forces a rebuild. I can detail this in comments.

Reply:
You mean if there is no changes in `args.gn` file but we just generate the new one forces the chip library rebuild, right?
Yes, detail comments will be appreciated.

Reply:
Comment added